src/components/button-icon-popover-control/index.js

```diff
@@ -80,14 +95,23 @@ class ButtonIconPopoverControl extends Component {
 		this.setState( { open: false } )
 	}
 
+	handleMouseLeave() {
+		this.setState( { isMouseOutside: true } )
+	}
+
+	handleMouseEnter() {
+		this.setState( { isMouseOutside: false } )
+	}
+
 	/**
 	 * Use our own click/close handler. Don't close when a popover (e.g. a colorpicker) is clicked.
-	 * If this is not used, the popover will close when a color control's custom color field (when inside the popover) is clicked.
+If this is not used, the popover will close when a color control's custom color field (when inside the popover) is clicked.
 	 *
-	 * @param {Event} ev Click event
+	 * @param ev
 	 */
 	handleOnClickOutside( ev ) {
-		if ( ev.relatedTarget && ! ev.relatedTarget.closest( '.components-popover' ) && ev.relatedTarget !== this.buttonRef.current ) {
+		const clickedElements = ev.path.filter( elements => elements.outerHTML === this.buttonRef.current.outerHTML )
+		if ( this.state.isMouseOutside && clickedElements.length === 0 ) {
```

Comment:
Nice try, but this is more readable and shorter:

```suggestion
		if ( this.state.isMouseOutside && ev.target.closest( 'button' ) !== this.buttonRef.current ) {
```